tests/js/client/communication/test-communication.js

```diff
@@ -470,96 +466,77 @@ function GenericAqlSetupPathSuite(type) {
     });
   `;
 
-  const apiLibs = `
-    const request = require("@arangodb/request");
-    arango.getEndpoint()
-    function sendRequest(method, endpoint, body, headers) {
-
-      const endpointToURL = (endpoint) => {
-        if (endpoint.substr(0, 6) === 'ssl://') {
-          return 'https://' + endpoint.substr(6);
-        }
-        var pos = endpoint.indexOf('://');
-        if (pos === -1) {
-          return 'http://' + endpoint;
-        }
-        return 'http' + endpoint.substr(pos);
-      };
-
-      let res;
-      try {
-        const envelope = {
-          json: true,
-          method,
-          url: endpointToURL(arango.getEndpoint()) + endpoint,
-          headers,
-        };
-        if (method !== 'GET') {
-          envelope.body = body;
-        }
-        res = request(envelope);
-      } catch (err) {
-        require("console").log(err);
-        return {};
-      }
-      if (typeof res.body === "string") {
-        if (res.body === "") {
-          res.body = {};
-        } else {
-          res.body = JSON.parse(res.body);
-        }
-      }
-      return res;
-    }
-  `;
   // Note: A different test checks that the API works this way
   const apiExclusive = `
-    ${apiLibs}
+  while (true) {
```

Comment:
its for having `return` abort, and break cut. Goto so to say.

Reply:
ah yes, now I see.
This is bad.
After this code is executed there is an insert into a collection, on which we wait
So in case of the error this insert is NOT happening anymore, causing the test to get stuck instead of fail.
We cannot do a return here, the code needs to complete until the end.